src/util/include/util/serializing/ObjectInputStream.h

```diff
@@ -52,4 +54,43 @@ class ObjectInputStream {
     size_t len = 0;
 };
 
-extern template void ObjectInputStream::readData(std::vector<double>& data);
+// This function requires that T is read from its binary representation to work (e.g. integer type)
+template <typename T>
+T readTypeFromSStream(std::istringstream& istream) {
+    if (istream.str().size() < sizeof(T)) {
```

Comment:
It is hard to get the size of a stream, therefore it should not be done at all, let it crash or check the stream flags.

Reply:
I have removed all use of `istream.str()`